source/Halibut.Tests/TcpClientCloseImmediatelyFixture.cs

```diff
@@ -0,0 +1,144 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using System.Net;
+using System.Net.Sockets;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Halibut.Tests.Transport.Streams;
+using Halibut.Tests.Util;
+using Halibut.Transport;
+using NUnit.Framework;
+
+namespace Halibut.Tests
+{
+    public class TcpClientCloseImmediatelyFixture : BaseTest
+    {
+        [Test]
```

Comment:
@acodrington and I talked about this.

This test is hard to write since `close()` and `CloseImmediately` appears to be fast even if the send buffer is full (and the last write timed out because the receiver is not reading any bytes).

Looking at ways to test this I notice some differences between `close()` and `CloseImmediatately`, when the read never reads and writes are done until a timeout is thrown:
| After the write times out and the one of the close methods its called: | Close()                | CloseImmediately()                                   | Reason                                                                                                                                                                |
|------------------------------------------------------------------------|------------------------|------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| When reading those writes the number of read bytes is                  | ~64k                   | ~32K                                                 | This probably happens because it reads both the bytes in some recieve buffer and in the case of just close() it also reads the bytes in the send buffer of the writer |
| When reading                                                           | No exception is thrown | An exception is thrown outlining a RST has occurred  |                                                                                                                                                                       |
|                                                                        |                        |                                                      |                                                                                                                                                                       |

Reply:
During our conversation, Luke and I figured out that the test would still pass even if we rolled back the change, so it is not providing any value as an automated test. This happens because in the old code, `TcpClient.Close()` would still return quickly, implying that the client and its socket were closed, but at the OS level, the send buffer was still keeping the socket open trying to send.

I tried reworking this test, but wasn't able to get a workable solution within a reasonable timebox.

I was planning to make the test keep its own copy of all the bytes that were sent by the client _and_ received by the server. Then it's a matter of running the test as before, but allowing the server to finish reading once the connection is closed i.e. finish reading whatever was already sitting in the received buffer. At this point we could compare the bytes sent vs bytes received, and if the connection was killed ASAP, the received bytes should be missing some that were sent, as they would've been dropped from the send buffer as part of the `CloseImmediately` call.

However, I was having to change the test significantly to allow the reading to continue until there was no more data to read from the buffer, and I hit my timebox.

@nathanwoctopusdeploy You mentioned that you were happy to rely on the manual verification for this functionality, so would you prefer I remove these tests? We could also keep them around with an `[ignore]` attribute, as they are still useful for manual verification with WireShark.